modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1401,7 +1400,23 @@ void ONNXImporter::populateNet(Net dstNet)
             }
             replaceLayerParam(layerParams, "mode", "interpolation");
 
-            if (layerParams.get<String>("interpolation") == "linear" && framework_name == "pytorch") {
+            //fused node
+            if (node_proto.input_size() == 1)
+            {
+                //unfused Resize subgraph fused to Upsample node
+                if (layerParams.has("coordinate_transformation_mode"))
+                {
+                    String interp_mode = layerParams.get<String>("coordinate_transformation_mode");
+                    CV_Assert_N(interp_mode != "tf_crop_and_resize", interp_mode != "tf_half_pixel_for_nn");
+
+                    layerParams.set("align_corners", interp_mode == "align_corners");
+                    if (layerParams.get<String>("interpolation") == "linear") {
+                        layerParams.set("interpolation", interp_mode == "pytorch_half_pixel" ?
+                                                "opencv_linear" : "bilinear");
+                    }
+                }
+            }
+            else if (layerParams.get<String>("interpolation") == "linear" && framework_name == "pytorch") {
```

Comment:
Can you please refactor so we first extract interpolation (without checking number of inputs) but separately scale factors? So make it more similar to `else if (layer_type == "Resize") above.

Reply:
okay.

Reply:
Please have a look if changes made are same as expected or you asked for something else.